src/query/storage/m3/storage.go

```diff
@@ -116,6 +102,36 @@ func (s *m3storage) Name() string {
 	return "local_store"
 }
 
+func (s *m3storage) FetchProm(
+	ctx context.Context,
+	query *storage.FetchQuery,
+	options *storage.FetchOptions,
+) (storage.PromResult, error) {
+	accumulator, err := s.fetchCompressed(ctx, query, options)
+	if err != nil {
+		return storage.PromResult{}, err
+	}
+
+	result, _, err := accumulator.FinalResultWithAttrs()
+	defer accumulator.Close()
```

Comment:
Why not `defer accumulator.Close()` before the call to `FinalResultWithAttrs` since it will run anyway (seems more normal to do a `defer foo.Close()` just after you have access to something rather than a few more lines down).

Reply:
Good call, changed in cleanup :+1: